rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -25,10 +26,28 @@
 
 from sawtooth_rest_api.protobuf.validator_pb2 import Message
 
-
 LOGGER = logging.getLogger(__name__)
 
 
+class _Backoff:
+    """Implements a simple exponential backoff mechanism.
+    """
+
+    def __init__(self):
+        self.num_retries_left = 3
+        self.min_interval = 0.1
+        self.max_interval = 1.0
+        self.interval = self.min_interval
+
+    async def do_backoff(self):
+        if not self.num_retries_left:
+            LOGGER.warning("No more retries left. Backoff terminated...")
```

Comment:
This looks like a duplicate of the other logging message below which starts with "Retry limit reached". Suggest removing this one and doing all logging outside this class.

Reply:
Looking at the comment below to include the retry number and backoff time, I think it'll be better to move all logging into this class and remove the logs outside.